### ISSUE TITLE
Video gets uploaded to report with videoUploadOnPasses = false in cypress.json

### ISSUE THREAD

**Issue by romankhomitskyi** [Shelex/cypress-allure-plugin#109]
**To Reproduce**
Video is uploaded to report even with option videoUploadOnPasses = false in cypress.json

**Comment by Shelex** [comment]
Hi @romankhomitskyi , thanks for reporting this issue. 
However, please note that `videoUploadOnPasses` - it is config for cypress to upload video to dashboard.
Allure plugin has its' own config for that - env variable `allureAddVideoOnPass`.

In v2.23.0 if `allureAddVideoOnPass` is false (it is false by default) video will not be attached for passed tests.

**Comment by romankhomitskyi** [comment]
> Hi @romankhomitskyi , thanks for reporting this issue. However, please note that `videoUploadOnPasses` - it is config for cypress to upload video to dashboard. Allure plugin has its' own config for that - env variable `allureAddVideoOnPass`.
> 
> In v2.23.0 if `allureAddVideoOnPass` is false (it is false by default) video will not be attached for passed tests.

Yeahh, you are right, anyway thanks for fix! Good job

**Comment by romankhomitskyi** [comment]
I have to reopen this issue 
Running this command

`yarn cypress run --browser chrome --env configFile=local,allure=true`

in my package.json

`"@shelex/cypress-allure-plugin": "^2.23.0"`

video is attached to report

![image](https://user-images.githubusercontent.com/57874507/145276146-82432a7c-b5f7-4048-a369-4afc82293204.png)

**Comment by romankhomitskyi** [comment]
@Shelex Hi. I was wondering if there is any progress so far?

**Comment by Shelex** [comment]
> @Shelex Hi. I was wondering if there is any progress so far?

Nope, had no time for that.

**Comment by romankhomitskyi** [comment]
@Shelex Hi. I was wondering if you could also handle on your side deleting videos? I have noticed that  after:spec event not working  in plugins.

```
allureWriter(on, config);
	on('after:spec', (spec, results) => {
		if (results && results.stats.failures === 0 && results.video) {
			// `del()` returns a promise, so it's important to return it to ensure
			// deleting the video is finished before moving on
			return del(results.video);
		}
	});
	return config;
```

It could be better move but not copy failed videos and delete passed ones

**Comment by Shelex** [comment]
@romankhomitskyi I am against removing files from plugin as they may be used by other plugin or be required for some user actions. I may consider doing that when some flag is passed.

> on('after:spec')
lol, I am afraid that cypress allows only one subscriber for this event ( it is used in plugin under the hood), so this should be investigated.

**Comment by romankhomitskyi** [comment]
> @romankhomitskyi I am against removing files from plugin as they may be used by other plugin or be required for some user actions. I may consider doing that when some flag is passed, but anyway: your artifacts are your artifacts, this plugin is responsible for gathering data only.
> 
> > on('after:spec')
> 
> lol, I am afraid that cypress allows only one subscriber for this event ( it is used in plugin under the hood), so this should be investigated.

Ok, thanks

**Comment by Shelex** [comment]
Hi @romankhomitskyi 
Issue is fixed in v2.24.0.

Regarding `on('after:spec')` event - unfortunately my assumption seems to be correct, cypress does not allow more than 1 listener for this event. Will be investigated in separate task.

**Comment by romankhomitskyi** [comment]
Thanks, for fix
